ddtrace/tracer/option.go

```diff
@@ -547,7 +547,9 @@ func newConfig(opts ...StartOption) *config {
 		if agentport := c.agent.StatsdPort; agentport > 0 && !c.agent.ignore {
 			// the agent reported a non-standard port
 			host, _, err := net.SplitHostPort(addr)
-			if err == nil {
+			// the code in the next if-block only makes sense if we are dealing with
+			// a user-defined TCP-based protocol URI.
```

Comment:
Your explanation below cleared up my confusion, so maybe we can include some of that info in this comment?
```suggestion
			// Use agent-reported address if it differs from the user-defined TCP-based protocol URI
```